pyteal/ast/itxn_test.py

```diff
@@ -134,4 +135,126 @@ def test_InnerTxnBuilder_SetFields():
                 expr.__teal__(teal4Options)
 
 
+ITXN_METHOD_CASES = (
+    (
+        pt.Int(1),
+        "add(uint64,uint64)void",
+        [pt.Itob(pt.Int(1)), pt.Itob(pt.Int(1))],
+        {TxnField.fee: pt.Int(0)},
+        pt.Seq(
+            pt.InnerTxnBuilder.SetFields(
+                {
+                    pt.TxnField.type_enum: TxnType.ApplicationCall,
+                    pt.TxnField.application_id: pt.Int(1),
+                    pt.TxnField.application_args: [
+                        pt.MethodSignature("add(uint64,uint64)void"),
+                        pt.Itob(pt.Int(1)),
+                        pt.Itob(pt.Int(1)),
+                    ],
+                    pt.TxnField.fee: pt.Int(0),
+                }
+            ),
+        ),
+    ),
+    (
+        pt.Int(1),
+        "add(uint64,uint64)void",
+        [pt.abi.Uint64(), pt.abi.Uint64()],
+        {TxnField.fee: pt.Int(0)},
+        pt.Seq(
+            pt.InnerTxnBuilder.SetFields(
+                {
+                    pt.TxnField.type_enum: TxnType.ApplicationCall,
+                    pt.TxnField.application_id: pt.Int(1),
+                    pt.TxnField.application_args: [
+                        pt.MethodSignature("add(uint64,uint64)void"),
+                        pt.abi.Uint64().encode(),
+                        pt.abi.Uint64().encode(),
```

Comment:
```suggestion
        [arg1 := pt.abi.Uint64(), arg2 := pt.abi.Uint64()],
        {TxnField.fee: pt.Int(0)},
        pt.Seq(
            pt.InnerTxnBuilder.SetFields(
                {
                    pt.TxnField.type_enum: TxnType.ApplicationCall,
                    pt.TxnField.application_id: pt.Int(1),
                    pt.TxnField.application_args: [
                        pt.MethodSignature("add(uint64,uint64)void"),
                        arg1.encode(),
                        arg2.encode(),
```

Referencing the same type instance in the expected response will strengthen the test. This is because it enables `TealBlock.MatchScratchSlotReferences` to check that the same underlying slots are used in the actual and expected responses.

Also applies to the next test with reference types.

(Writing this in a browser to convey the general idea, I haven't tested it)